### PR TITLE
Node develop

### PR DESCRIPTION
**Pull request by Mihail-tech**

### CODE REVIEW

File: bin/www [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
```
Comment (k-krasilnikova) [inline bin/www:1]: Add to gitignore

File: app.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
```
Comment (k-krasilnikova) [inline app.js]: replace with `let` or `const` according to context

File: app.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
```
Comment (k-krasilnikova) [inline app.js]: Replace with `import`s please

File: app.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+app.set('view engine', 'jade');
+
+app.use(logger('dev'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.use('/api/greetings', indexRouter);
+app.use('/users', usersRouter);
```
Comment (k-krasilnikova) [inline app.js:25]: Do we really needthis handler?

File: routes/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+var express = require("express");
```
Comment (k-krasilnikova) [inline routes/index.js]: import

File: routes/index.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+var router = express.Router();
```
Comment (k-krasilnikova) [inline routes/index.js]: const

File: routes/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+/* GET home page. */
+router.get("/", (req, res) => {
+  const name = req.query.name;
```
Comment (k-krasilnikova) [inline routes/index.js:6]: Can you please replace it with an object destructuring?

File: routes/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  if (name === undefined) {
```
Comment (k-krasilnikova) [inline routes/index.js:7]: We can check it just like:
`if (name) {`

File: routes/index.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    res.status(400).json({ name: "null", message: "You should write name" });
+  } else {
+    res.render("index", { name });
```
Comment (k-krasilnikova) [inline routes/index.js]: What we will render in case of empty name (empty string in param)?

File: routes/notes.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+const express = require('express')
```
Comment (k-krasilnikova) [inline routes/notes.js]: import

File: routes/users.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
+var router = express.Router();
+
+/* GET users listing. */
+router.get('/', function(req, res, next) {
```
Comment (k-krasilnikova) [inline routes/users.js]: remove this file

File: views/error.jade [hunk 12]
```diff
@@ -0,0 +1,6 @@
+extends layout
+
```
Comment (k-krasilnikova) [inline views/error.jade]: remove this file

File: views/index.jade [hunk 13]
```diff
@@ -0,0 +1,4 @@
+extends layout
```
Comment (k-krasilnikova) [inline views/index.jade]: remove this file

File: views/layout.jade [hunk 14]
```diff
@@ -0,0 +1,7 @@
+doctype html
+html
```
Comment (k-krasilnikova) [inline views/layout.jade]: remove this file

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Some changes required!

File: routes/index.js (hunk 9 above)
Comment (Mihail-tech) [reply routes/index.js]: we will see mistake because page write only when we do name

**Review by k-krasilnikova — APPROVED** [review]
Good job!

**Pull request merged**